internal/auth/options.go

```diff
@@ -129,20 +131,82 @@ type SignatureData struct {
 	key  string
 }
 
-// NewOptions returns new options
-func NewOptions() *Options {
-	return &Options{
-		Port:            4180,
-		CookieName:      "_sso_auth",
-		CookieSecure:    true,
-		CookieHTTPOnly:  true,
-		CookieExpire:    time.Duration(168) * time.Hour,
-		CookieRefresh:   time.Duration(0),
-		SetXAuthRequest: false,
-		PassUserHeaders: true,
-		PassHostHeader:  true,
-		ApprovalPrompt:  "force",
-		RequestLogging:  true,
+// NewOptions returns new options with the below overrides
+func NewOptions() (*Options, error) {
+	overrides := map[string]interface{}{
```

Comment:
It seems strange to me that we need both these as well as the defaults. Is there any way we could consolidate the two?

Reply:
We don't need both of these - this was just in order to mimc the previous behaviour of having default settings, but also hard overrides. (unless I misunderstood that behaviour)

Something I wanted to bring up to question anyway as I wasn't completely sure of the motivation behind it. Very happy to consolidate if there isn't any need.

Reply:
If it's easy, let's try to consolidate.